### PR TITLE
tools: reduce diff noise with NetRadiant

### PR DESCRIPTION
**Pull request by illwieckz**
This a PR to reduce diff noise between GtkRadiant and NetRadiant in the `tools` tree.

There is full of slightly minor change like whitespaces or newlines or headers or sometime just another way to do things because two people fixed the same issue at the same time on both side and they just not written the same line in the same order. All of this makes cherry-picking across forks annoying because it raises stupid merge conflict about things that are not meaningful.

Hence this Merge Request.

I also imported five fixes from NetRadiant side.

### CODE REVIEW

**Comment by illwieckz** [comment]
See [netradiant!82](https://gitlab.com/xonotic/netradiant/merge_requests/82) for a similar merge request on NetRadiant side.

File: tools/quake3/common/polylib.c [hunk 1]
```diff
@@ -461,15 +461,15 @@ winding_t *BaseWindingForPlane( vec3_t normal, vec_t dist ){
...
+	size_t size;
 	winding_t   *c;
 
 	if ( !w ) {
 		Error( "CopyWinding: winding is NULL" );
 	}
 
 	c = AllocWinding( w->numpoints );
-	size = sizeof( *w ) + sizeof( *w->p ) * w->numpoints;
+	size = (size_t)sizeof( *w ) + sizeof( *w->p ) * w->numpoints;
```
Comment (tkoeppe) [inline tools/quake3/common/polylib.c]: This seems a bit pointless. The type of a `sizeof` expression is already `size_t`.

File: tools/quake3/common/threads.c [hunk 2]
```diff
@@ -551,7 +551,7 @@ void RunThreadsOn( int workcnt, qboolean showpacifier, void ( *func )( int ) ){
 		for ( i = 0 ; i < numthreads ; i++ )
 		{
 			/* Default pthread attributes: joinable & non-realtime scheduling */
-			if ( pthread_create( &work_threads[i], NULL, (void*)func, (void*)i ) != 0 ) {
+			if ( pthread_create( &work_threads[i], NULL, (void*)func, (void*)(size_t)i ) != 0 ) {
```
Comment (tkoeppe) [inline tools/quake3/common/threads.c]: `intptr_t` would be slightly more appropriate here: it's an integral type guaranteed to be able to store any object pointer.

File: tools/quake3/common/vfs.c [hunk 3]
```diff
@@ -289,6 +289,7 @@ int vfsLoadFile( const char *filename, void **bufferptr, int index ){
 
 		f = fopen( filename, "rb" );
 		if ( f == NULL ) {
+			fclose(f);
```
Comment (tkoeppe) [inline tools/quake3/common/vfs.c]: I'm pretty sure it's undefined behaviour to call `fclose` on a null pointer. That's not to speak of the fact that it's completely pointless to attempt to close a file that was never opened.

**Comment by tkoeppe** [comment]
Hm, these changes appear to have introduced some bugs and some noise. I left individual comments.

File: tools/quake3/common/vfs.c (hunk 3 above)
Comment (illwieckz) [reply tools/quake3/common/vfs.c]: hmm, that's right!

File: tools/quake3/common/polylib.c (hunk 1 above)
Comment (illwieckz) [reply tools/quake3/common/polylib.c]: thanks, I will report it on netradiant side too (the point is also to have the same thing on both side, but it must makes sense yes)

File: tools/quake3/common/threads.c (hunk 2 above)
Comment (illwieckz) [reply tools/quake3/common/threads.c]: I'm doing it!

**Pull request merged**